### ISSUE TITLE
Unc0ver seems crashed at (6/40) while trying to JB iPad Mini 3 11.1.1

### ISSUE THREAD

**Issue by liuxuan30** [pwn20wndstuff/Undecimus#219]
### UPDATE
Device info:
iPad mini 3, A1599, iOS 11.1.1 (15B150)

I installed the app from source and debug, it shows:
```
2019-01-08 11:32:50.537873+0800 Undecimus[220:4773] [*] PROGRESS: Exploiting (6/40)
2019-01-08 11:32:50.538064+0800 Undecimus[220:4773] [*] Initializing QiLin...
Here - Kernel Base: 0xfffffff008804000
2019-01-08 11:32:50.545356+0800 Undecimus[220:4773] ERROR: Unable to compress kernelcache (is it still LZSS?) - this is not fatal, but means you have to hardcode symbols until J updates.. Sorry.
2019-01-08 11:32:50.545643+0800 Undecimus[220:4773] STATUS: Loading symbols...
(lldb) 
```
and crashed at
>_assert(initQiLin(tfp0, kernel_base) == ERR_SUCCESS, message, true);

### The context
I have a Mini 3 previously as my dev device to test my tweak, using Electra (1.0.4 I suppose)

I wanted to switch to unc0ver 2.1.0, so I ran SemiRestore11-Lite and the `unjailbreak.sh` script from https://github.com/KirovAir/delectra/blob/master/unjailbreak.sh

Everything seems working just fine, and then I installed Unc0ver via impactor, **use all default settings**, and just hit jailbreak, but the app keeps like crashing in (1/40) or (6/40), and no any prompt nor error message.

I tried the 2.1.1 beta version, same issue. Is this a bug or my device messed up? I used the same steps on my iPhone, and unc0ver works just fine, so I continued with my iPad.

Device info:
iPad mini 3, A1599, iOS 11.1.1 (15B150)

log while cleaning electra:
```
iOS-24:/usr/lib/TweakInject root# SemiRestore11-Lite 
Performing Stage 2.1.
Performing Stage 2.2.
Performing Stage 2.3.
(Reading database ... 3583 files and directories currently installed.)
Removing adv-cmds (172) ...
Removing basic-cmds (55) ...
Removing bootstrap-cmds (98) ...
Removing co.dynastic.tsssaver (1.2.0) ...
Removing com.mytweak.mi.worker (2.0.1-7+debug) ...
dpkg: warning: while removing com.mytweak.mi.worker, directory '/private/var/root' not empty so not removed
Removing com.mytweak.mianalysis (2.0.1-2+debug) ...
Removing com.autopear.installipa (3.4.1-1+debug) ...
Removing com.conradkramer.open (1.1.1-3+debug) ...
Removing com.ericasadun.utilities (1:0.4.2) ...
Removing com.modmyi.libswift4 (4.2-1) ...
Removing coreutils (8.29-1) ...
Removing developer-cmds (63) ...
Removing diskdev-cmds (593-1) ...
Removing file-cmds (272) ...
Removing gawk (4.2.1) ...
Removing git (2.16.2) ...
Removing iokittools (108) ...
Removing ldid (2:1.2.2-coolstar) ...
Removing less (530) ...
Removing lsof (62-1) ...
Removing make (4.2.1) ...
Removing mobilesubstrate (99.0) ...
Removing nano (2.9.7) ...
Removing network-cmds (543) ...
Removing org.coolstar.semirestore11-lite (11.0) ...
Removing org.coolstar.tweakinject (1.0.7-2) ...
Removing p7zip (16.02) ...
Removing patch (2.7.6) ...
Removing rsync (3.1.3) ...
Removing syslogcmdline (1.0) ...
Removing top (111.20.1) ...
Removing unrar (5.5.8) ...
Removing unzip (6.0) ...
Removing vim (8.0.1848) ...
Removing wget (1.19) ...
Removing zip (3.0) ...
Removing com.ex.libsubstitute (0.0.7-coolstar) ...
Removing curl (7.59.0-1) ...
Removing inetutils (1.9.4) ...
Removing nghttp2 (1.31.0) ...
shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
dpkg: warning: ignoring request to remove com.coolstar.stash933 which isn't installed
shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
iOS-24:/usr/lib/TweakInject root# cd /electra/
iOS-24:/electra root# ls
amfid_payload.dylib*  inject_criticald*  jailbreakd_client*   pspawn_payload.dylib*
helloworld*           jailbreakd*        libjailbreak.dylib*  unjailbreak.sh
iOS-24:/electra root# sh unjailbreak.sh 
Electra uninstaller by KirovAir for Electra 1.0.x
This has ONLY been tested on a few devices and might kill your JB.
Assuming you have not installed any other jailbreak or modified the rootfs directly yourself, you should be on stock iOS once this is complete.
Press enter to continue. Press Ctrl + C to exit
Killing Cydia..
No matching processes were found
Removing exploit generated..
Removing potential manual files..
Removing electra beta files..
Removing Cydia bootstrap.. (Please pray from here onwards)
Removing electra bootstrap..
Removing launchctl and jailbreakd process leftovers..
Removing possible topanga junk (just to make sure you can rejailbreak in case of emergency)
Fixing up hosts file..
Removing last tools and clearing ui cache..
All jailbreak related files are removed! (At least all included in the iOS 11 electra JB)
Rebooting... (Keep praying my friend!)
iOS-24:/electra root# Connection to 192.168.6.24 closed by remote host.
Connection to 192.168.6.24 closed.
Transferred: sent 4684, received 11836 bytes, in 206.7 seconds
Bytes per second: sent 22.7, received 57.3
```

**Comment by pwn20wndstuff** [comment]
Ok I will take a look at this.

**Comment by TheGeneralNeilD** [comment]
Pwn, I did drop you a message, I had this very same problem, I have to go back to the 2.1 pre-release (the one that had 50 steps and run this first, then use the release (40 steps) all was ok then
I use MP version signed by myself if thats any help

**Comment by pwn20wndstuff** [comment]
@liuxuan30 Can you try moving initQiLin() out of _assert() to remove the assertion and try it again to see if the rest of the jailbreak succeeds?

**Comment by liuxuan30** [comment]
> @liuxuan30 Can you try moving initQiLin() out of _assert() to remove the assertion and try it again to see if the rest of the jailbreak succeeds?

What code base  do you need me to? My current code base or latest master branch ?

**Comment by pwn20wndstuff** [comment]
@liuxuan30 Latest master

**Comment by Cryptiiiic** [comment]
duplicate of #65

**Comment by liuxuan30** [comment]
@sbingner I could try now. Update later

**Comment by liuxuan30** [comment]
@sbingner not working on my iPad. still same symptom.

**Comment by Cryptiiiic** [comment]
Should be fixed now. Qilin was removed.
